plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -318,14 +346,18 @@ func TestGetCallRates(t *testing.T) {
 			params := metricsstore.CallRateQueryParameters{
 				BaseQueryParameters: buildTestBaseQueryParametersFrom(tc),
 			}
+			tracer, exp, closer := tracerProvider(t)
 			cfg := defaultConfig
 			if tc.updateConfig != nil {
 				cfg = tc.updateConfig(cfg)
 			}
-			reader, mockPrometheus := prepareMetricsReaderAndServer(t, cfg, tc.wantPromQlQuery, nil)
+			reader, mockPrometheus := prepareMetricsReaderAndServer(t, cfg, tc.wantPromQlQuery, nil, tracer)
 			defer mockPrometheus.Close()
 
 			m, err := reader.GetCallRates(context.Background(), &params)
+			assert.NotEmpty(t, exp.GetSpans(), "Spans recorded during the test.")
```

Comment:
NotEmpty check is redundant when followed by Len=1 check.

Reply:
That means I have to rmv every assert NotEmpty check.